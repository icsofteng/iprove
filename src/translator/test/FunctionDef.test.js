const {translate} = require('../../translator/z3')

const test_constants = ['p']
const test_rules = [
  {
    dependencies: [],
    ast: {
      type: 'function-def',
      name: 'F',
      inputTypes: [
        {
          type: 'varType',
          value: 'Int'
        },
        {
          type: 'varType',
          value: 'Int'
        }
      ],
      outputType: {
        type: 'varType',
        value: 'Int'
      }
    }
  },
  {
    dependencies: [],
    ast: {
      type: 'literal',
      value: 'p'
    }
  }
]

test('Function Def test', () => {
  expect(translate(test_rules, test_constants, [])).toMatchSnapshot()
})