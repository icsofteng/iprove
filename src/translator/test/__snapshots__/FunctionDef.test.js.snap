// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function Def test 1`] = `
"(declare-sort Type)
(declare-const p Bool)
(declare-fun F (Int Int ) Int)
(assert (not p))
(check-sat)"
`;
